frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,224 @@
+  },
+];
+
+export const generateWarningTitle = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
+): string => {
+  if (!isEnabled) {
+    return 'All hardware profiles are disabled';
+  }
+  if (isAllWarnings) {
+    return 'All hardware profiles are invalid';
+  }
+  if (warningCount > 0) {
+    return 'One or more hardware profiles are invalid';
+  }
+  return '';
+};
+
+export const generateWarningMessage = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
+): string => {
+  if (!isEnabled) {
+    return 'You must have at least one hardware profile enabled for users to create workbenches or deploy models. Enable one or more profiles in the table below.';
+  }
+  if (isAllWarnings) {
+    return 'You must have at least one valid hardware profile enabled for users to create workbenches or deploy models. Take the appropriate actions below to re-validate your profiles.';
+  }
+  if (warningCount > 0) {
+    return 'One or more of your defined hardware profiles are invalid. Take the appropriate actions below to revalidate your profiles.';
+  }
+  return '';
+};
+
+export const generateWarningForHardwareProfiles = (
+  hardwareProfiles: HardwareProfileKind[],
+): HardwareProfileWarningMessage => {
+  let isEnabled = false;
+  let warningCount = 0;
+  for (const profile of hardwareProfiles) {
+    if (profile.spec.warning?.warningStatus === true) {
+      warningCount += 1;
+    }
+    if (profile.spec.enabled === true) {
+      isEnabled = true;
+    }
+  }
+
+  return {
+    warning: warningCount > 0 || !isEnabled,
+    title: generateWarningTitle(isEnabled, warningCount === hardwareProfiles.length, warningCount),
+    message: generateWarningMessage(
+      isEnabled,
+      warningCount === hardwareProfiles.length,
+      warningCount,
+    ),
+    isEnabled,
+  };
+};
+
+export const hardwareProfileWarning = (
+  hardwareProfiles: HardwareProfileKind[],
+): HardwareProfileKind[] =>
+  hardwareProfiles.map((profile) => {
+    let warningStatus = false;
+    const { identifiers } = profile.spec;
+    let parentWarningMessage = '';
+    const isDefaultProfile = isHardwareProfileOOTB(profile);
+    if (isDefaultProfile && !hasCPUandMemory(identifiers ?? [])) {
+      for (let identifier of identifiers ?? []) {
+        if (
+          identifier.resourceType === IdentifierResourceType.CPU ||
+          identifier.resourceType === IdentifierResourceType.MEMORY
+        ) {
+          identifier = {
+            ...identifier,
+            warning: true,
+          };
+        }
+      }
+      warningStatus = true;
+      parentWarningMessage =
+        'Select the restore button to restore the Default profile to its initial state.';
+    } else {
+      for (let identifier of identifiers ?? []) {
+        try {
+          if (identifier.minCount.toString().at(0) === '-') {
+            parentWarningMessage = `Minimum allowed ${
+              identifier.resourceType ?? 'resource label'
```

Comment:
I saw a lot of `identifier.resourceType ?? 'resource label'` in this file. Are you trying to say `identifier.displayName`? I believe you can replace all the places with `identifier.displayName` no matter what type is it. (And maybe add a <b> tag?)